diff --git a/MediaInfoLib/Source/MediaInfo/File__Analyze_Buffer.cpp b/MediaInfoLib/Source/MediaInfo/File__Analyze_Buffer.cpp
index 4f418b6..d8c86df 100644
--- a/MediaInfoLib/Source/MediaInfo/File__Analyze_Buffer.cpp
+++ b/MediaInfoLib/Source/MediaInfo/File__Analyze_Buffer.cpp
@@ -82,7 +82,6 @@ extern MediaInfo_Config Config;
 //***************************************************************************
 // Init
 //***************************************************************************
-
 //---------------------------------------------------------------------------
 void File__Analyze::BS_Begin()
 {
@@ -97,7 +96,32 @@ void File__Analyze::BS_Begin()
         BS_Size=Buffer_Size-(size_t)(Buffer_Offset+Element_Offset);
     else
         BS_Size=0;
-
+    
+    //BS_Begin Buffer=0xc2690db0 Buffer_Offset=0x0 Element_Offset=0x0 BS_Size=0x27  
+    int ii=0;
+    int8u Info;
+    printf("BS_Begin Buffer=0x%llx Buffer_Offset=0x%x Element_Offset=0x%x BS_Size=0x%x  \n",Buffer,Buffer_Offset,Element_Offset,BS_Size);
+    if(BS_Size==0x27)
+    {
+        //Info=BS->Get8(64);
+        //printf("Info=0x%x Offset_Get=0x%lx Remain()=0x%lx \n",Info,BS->Offset_Get(),BS->Remain());
+        //Info=BS->Get8(64);
+        //printf("Info=0x%x Offset_Get=0x%lx Remain()=0x%lx \n",Info,BS->Offset_Get(),BS->Remain());
+        printf("BS_Begin dump extension Buffer:\n ");
+        printf("        0x00 0x01 0x02 0x03 0x04 0x05 0x06 0x07 0x08 0x09 0x0a 0x0b 0x0c 0x0d 0x0e 0x0f\n");
+        printf("-----------------------------------------------------------------------------------");
+        for(ii=0;ii<BS_Size;ii++)
+        {
+            if(ii%16==0)
+            {
+                printf("\n0x%02x   : ",(ii/16)*16);
+            }
+            printf("0x%02x ",*((unsigned char *)(Buffer+Buffer_Offset+(size_t)Element_Offsetii)));
+        }
+        //printf("*\nBuffer=0x%llx\n",*((char *)Buffer));
+        printf("\n");
+    }
+    
     BS->Attach(Buffer+Buffer_Offset+(size_t)(BS_Size?Element_Offset:0), BS_Size);
     #if MEDIAINFO_TRACE
         BS_Size<<=3; //In bits
@@ -2115,6 +2139,10 @@ void File__Analyze::Get_S1(int8u Bits, int8u &Info, const char* Name)
 {
     INTEGRITY_INT(Bits<=BS->Remain(), "Size is wrong", BS->Offset_Get())
     Info=BS->Get1(Bits);
+    if(strcmp(Name,"matrix_coefficients")==0)
+    {
+        printf("Name=%s Info=0x%x Bits=%d Offset_Get=0x%lx Remain()=0x%lx \n",Name,Info,Bits,BS->Offset_Get(),BS->Remain());
+    }
     #if MEDIAINFO_TRACE
         if (Trace_Activated)
         {
diff --git a/MediaInfoLib/Source/MediaInfo/Video/File_Mpegv.cpp b/MediaInfoLib/Source/MediaInfo/Video/File_Mpegv.cpp
index 06f6530..5543675 100644
--- a/MediaInfoLib/Source/MediaInfo/Video/File_Mpegv.cpp
+++ b/MediaInfoLib/Source/MediaInfo/Video/File_Mpegv.cpp
@@ -3715,7 +3715,7 @@ void File_Mpegv::extension_start()
 {
     Element_Name("extension_start");
     MPEG_Version=2; //extension_start only exists in MPEG-2 specs
-
+	//ext_flag = 1;
     //Parsing
     int8u extension_start_code_identifier;
     BS_Begin();
diff --git a/ZenLib/Source/ZenLib/BitStream_Fast.h b/ZenLib/Source/ZenLib/BitStream_Fast.h
index 8f579f3..e9d5b7b 100644
--- a/ZenLib/Source/ZenLib/BitStream_Fast.h
+++ b/ZenLib/Source/ZenLib/BitStream_Fast.h
@@ -40,6 +40,7 @@ public:
 
     void Attach(const int8u* Buffer_, size_t Size_)
     {
+        //printf("Fast Attach \n");
         Buffer=Buffer_;
         Buffer_Size=Buffer_Size_Init=Size_*8; //Size is in bits
         BufferUnderRun=false;
@@ -95,6 +96,7 @@ public:
         else
             ToReturn=LastByte<<NewBits;
         LastByte=*Buffer;
+        //printf("Buffer=0x%x Buffer=0x%x Buffer=0x%x  Buffer=0x%x \n",*Buffer,*(Buffer+1),*(Buffer+2),*(Buffer+3)); //如果Get_S1与BS_Begin dump buffer差异过大，可放开此处确认实际offset上的值
         Buffer++;
         Buffer_Size-=HowMany;
         ToReturn|=(LastByte>>(Buffer_Size%8))&Mask[NewBits];
